04-vue-cli/02-TheToaster/components/UiToast.vue

```diff
@@ -0,0 +1,30 @@
+<template>
+  <div class="toast" :class="className">
+      <UiIcon class="toast__icon" :icon="icon" />
+      <span>{{ message }}</span>
+    </div>
+</template>
+
+<script>
+import UiIcon from './UiIcon.vue';
+
+export default {
+  name: 'UiToast',
+
+  props: {
+    message: {
+      type: String
+    },
+    className: {
+      type: String
+    },
+    icon: {
+      type: String
+    }
```

Comment:
Usually, there is no need for a `className` prop for a root element class. We can set root element classes directly by a `class` attribute while using any components. `class*` props are only used when there is a need to pass some custom styles to an inner element.

Reply:
But in such components, it is better to pass `type` of the toast (`success`, `error`) rather than its icon or class. Users of the component should not manually define styles to make different types of toasts if they are pre-defined by design.